bokulich-lab/q2-fondue
# ENH: introduce NCBIAccessionIDs type

This PR introduces a new `NCBIAccessionIDs` type to allow keeping track of metadata files (and included accession IDs) in QIIME's provenance.

**Summary of changes:**
- a data import step is required before running data fetching actions to import metadata _tsv_ file into a QIIME artifact
- it won't be possible anymore to use plain _tsv_ files as ID input to `get-*` actions
- `SRAFailedIDs` and `SRAMetadata` are the only other types that are allowed as input
- except of the things described above, everything else works as before
- an added benefit: ID validation can now be performed when user imports the data and can be expanded by other ID types in the future, if needed

**To test this PR:**
- import your ID list as described in the README
- run any/all of the `get-*` actions with the generated artifact
- try to re-run the actions with generated `SRAMetadata` and, if possible, `SRAFailedIDs` artifacts
- `qiime metadata tabulate` the obtained metadata artifact and look at the provenance graph - you should see an MD5 hash of the original _tsv_ file (see an example below)

![Screen Shot 2022-01-28 at 09 08 29](https://user-images.githubusercontent.com/22281933/151510527-fb9ec4c4-0ebf-4855-8dc4-c642daad3480.png)

Closes #87.
